server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/model/assertion/AssertionTests.java

```diff
@@ -0,0 +1,486 @@
+import static java.lang.Math.log;
+import static java.lang.Math.max;
+import static org.testng.Assert.assertEquals;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import org.testng.annotations.DataProvider;
+import us.freeandfair.corla.model.CastVoteRecord;
+import us.freeandfair.corla.model.CastVoteRecord.RecordType;
+
+/**
+ * This class does not contain any tests, but provides utilities for use by the
+ * NEB/NEN Assertion test classes. Although much of the assertion-related functionality is
+ * present in the abstract Assertion class, we have chosen to test this functionality
+ * twice -- once in the context where the assertion is an NEB and then in the context
+ * where the assertion is an NEN. Although, at present, it is the same code that is
+ * being tested in both cases, it is possible that future development may alter the
+ * assertion subclasses in potentially unforeseen ways. This extra level of testing will
+ * reduce the likelihood of future programming errors if this eventuates.
+ */
+public class AssertionTests {
+
+  /**
+   * Constant representing a test contest name.
+   */
+  public static final String TC = "Test Contest";
+
+  /**
+   * A 10% risk limit, as a BigDecimal.
+   */
+  public static final BigDecimal riskLimit10 = BigDecimal.valueOf(0.10);
+
+  /**
+   * A 5% risk limit, as a BigDecimal.
+   */
+  public static final BigDecimal riskLimit5 = BigDecimal.valueOf(0.05);
+
+  /**
+   * A small raw margin of 10 votes.
+   */
+  public static final Integer smallRawMargin = 10;
+
+  /**
+   * A small diluted margin of 0.001.
+   */
+  public static final BigDecimal smallMargin = BigDecimal.valueOf(0.001);
+
+  /**
+   * A high assertion difficulty value of 100.
+   */
+  public static final BigDecimal highDifficulty = BigDecimal.valueOf(100);
+
+  /**
+   * A large raw margin of 10000 votes.
+   */
+  public static final Integer largeRawMargin = 10000;
+
+  /**
+   * A large diluted margin of 0.55.
+   */
+  public static final BigDecimal largeMargin = BigDecimal.valueOf(0.55);
+
+  /**
+   * A small assertion difficulty of 1.
+   */
+  public static final BigDecimal smallDifficulty = BigDecimal.valueOf(1);
+
+  /**
+   * List of candidates called "W", "L", and "O".
+   */
+  public static final List<String> wlo = List.of("W", "L", "O");
+
+  /**
+   * A parameter set consisting of two audited ballot record types (AUDITOR_ENTERED and REAUDITED).
+   * @return An array of parameter arrays.
+   */
+  @DataProvider(name = "AuditedRecordTypes")
+  public static Object[][] auditedRecordTypes(){
+    return new Object[][]{
+        {RecordType.AUDITOR_ENTERED},
+        {RecordType.REAUDITED}
+    };
+  }
+
+  /**
+   * Returns a set of varying parameters to supply when constructing assertions to test
+   * sample size computations. To test sample size calculation, we consider the following dimensions:
+   * -- Risk limit (5% or 10%)
```

Comment:
It doesn't really matter, but I believe CO typically uses 3%.

Reply:
It may make more sense to run all the tests with 3% and 5%, forgetting about 10%